index.js

```diff
@@ -88,27 +88,54 @@ NodePreGypGithub.prototype.uploadAsset = function(cfg){
 		filePath: cfg.filePath
 	}, function(err){
 		if(err) {console.error(err); return;}
-		console.log('Staged file ' + cfg.fileName + ' saved to ' + this.owner + '/' +  this.repo + ' release ' + this.package_json.version + ' successfully.');
+		console.log('Staged file ' + cfg.fileName + ' saved to ' + this.owner + '/' +  this.repo + ' release ' + this.release.tag_name + ' successfully.');
 	}.bind(this));
 };
 
-NodePreGypGithub.prototype.uploadAssets = function(){
-	var asset;
-	console.log("Stage directory path: " + path.join(this.stage_dir));
-	fs.readdir(path.join(this.stage_dir), function(err, files){
-		if(typeof files === 'undefined') {console.log('no files found'); return;}
-		files.forEach(function(file){
+function asyncFlatMap(list, iterator, callback) {
+	var result = [], left = list.length;
+	if (!left) return callback(result);
+	list.forEach(function(item) {
+		iterator(item, function(item) {
+			if (item != null) result = result.concat(item);
+			if (!--left) callback(result);
+		});
+	});
+}
+
+function findFilesDeep(dir, callback) {
+	fs.readdir(dir, function(err, names) {
+		asyncFlatMap(names || [], function(name, next) {
+			var file = path.join(dir, name);
+			fs.stat(file, function(err, stat) {
+				if (stat) {
+					if (stat.isFile()) { return next(file); }
+					else if (stat.isDirectory()) { return findFilesDeep(file, next); }
+				}
+				next();
+			});
+		}, callback);
+	});
+};
+
+NodePreGypGithub.prototype.uploadAssets = function() {
+	console.log("Stage directory path: " + this.stage_dir);
+	findFilesDeep(this.stage_dir, function(paths) {
```

Comment:
I guess you are right. To determine `stage_dir` we need an information from package.json so line 17 is to early, however after [this](https://github.com/bchr02/node-pre-gyp-github/blob/master/index.js#L119) line we could add:

```js
options.tag_name = options.tag_name || this.package_json.version;
if (this.package_json.binary.remote_path) {
  this.stage_dir = path.join(this.stage_dir, options.tag_name);
}
```

and at the same time if user provides incorrect tag_name or somehow mangles the remote_path the later code would not be able to find files. So the added benefit of this solution is configuration sanity checking.

I would also check if tag_name contains all the correct characters and threw otherwise.

Let me know what you think.